src/modules/fancyzones/lib/JsonHelpers.cpp

```diff
@@ -438,6 +467,7 @@ namespace JSONHelpers
 
     void FancyZonesData::LoadFancyZonesData()
     {
+        std::scoped_lock lock{ dataLock };
```

Comment:
Edit: forget about this, I forgot that the mutex is recursive.

**A deadlock will happen** if you end up calling SaveFancyZonesData here!
This also applies to all places where you call one FancyZonesData method from another and they both use locking.